sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/EnumerableExtensions.cs

```diff
@@ -72,6 +72,12 @@ public static IEnumerable<T> WhereNotNull<T>(this IEnumerable<T> enumerable)
             return enumerable.Where(e => e != null);
         }
 
+        public static IEnumerable<T> WhereNotNull<T>(this IEnumerable<Nullable<T>> enumerable)
+                where T : struct
+        {
+            return enumerable.Where(x => x.HasValue).Select(x => x.Value);
```

Comment:
_minor_ can be an arrow function

Reply:
I find it difficult to read with the `where` clause for optical reasons.
But I've applied this on the JoinStr below that I've recently added.